extraction_function.go

```diff
@@ -0,0 +1,4 @@
+package godruid
+
+type ExtractionFn interface{}
```

Comment:
can we not make it more specific than just an interface{}?

Reply:
at the very least it needs a `GetType() string`. "lower" for example. most extraction functions take arguments, but that one doesn't.

Reply:
changed it to a struct